augur/filter.py

```diff
@@ -1051,6 +1052,26 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
                     })
                 # month = (year, month)
                 metadata['month'] = list(zip(metadata['year'], metadata['month']))
+            if 'week' in generated_columns_requested:
+                # skip ambiguous days
+                df_skip = metadata[metadata['day'].isnull()]
+                metadata.dropna(subset=['day'], inplace=True)
+                for strain in df_skip.index:
+                    skipped_strains.append({
+                        "strain": strain,
+                        "filter": "skip_group_by_with_ambiguous_day",
+                        "kwargs": "",
+                    })
+                # Note that week = (year, week) from the date.isocalendar().
+                # Do not combine the raw year with the ISO week number alone,
+                # since raw year ≠ ISO year.
+                metadata['week'] = metadata.apply(lambda row: get_iso_year_week(row['year'], row['month'], row['day']), axis=1)
+
+                # If both 'week' and 'year' are requested, ignore 'year' since
+                # ISO year should take precedence and is already implied with
+                # 'week'.
+                if 'year' in generated_columns_requested:
+                    metadata['year'] = pd.NA
```

Comment:
I feel like this can potentially lead to some weird bugs...

Could we move the logic for "ISO year from 'week' takes precedence over 'year'" further up the function? 
i.e. if `group_by` includes both "year" and "week", just remove "year" from the list?